### PR TITLE
Disallow old style octal literal types

### LINKED ISSUE #10101: Old-style octal literal types are treated as decimal literal types
``` ts
// @target: es3
const i: 010 = 8;
```

Actual:

```ts
const i: 010 = 8;
      ~
      Error: Type '8' is not assignable to type '10';
```

Expected:

```ts
const i: 010 = 8;
         ~~~
         Error: Octal literal types must use ES2015 syntax. Use the syntax '0o10'.
```

### PR DESCRIPTION
**Pull request by arusakov**
Fixes #10101

### CODE REVIEW

**Comment by arusakov** [comment]
Waiting for master green build

**Comment by andy-ms** [comment]
Will need to rebase onto #13025 (when it's merged) to fix tests.

File: tests/baselines/reference/literals.errors.txt [hunk 1]
```diff
@@ -36,14 +36,14 @@ tests/cases/conformance/expressions/literals/literals.ts(25,10): error TS1085: O
...
-!!! error TS1085: Octal literals are not available when targeting ECMAScript 5 and higher.
+!!! error TS1085: Octal literals are not available when targeting ECMAScript 5 and higher. Use the syntax '0o1'.
     var n = 0x1;
     var n = -1;
     var n = -1.0;
     var n = -1e-4;
     var n = -003; // Error in ES5
              ~~~
-!!! error TS1085: Octal literals are not available when targeting ECMAScript 5 and higher.
+!!! error TS1085: Octal literals are not available when targeting ECMAScript 5 and higher. Use the syntax '0o3'.
```
Comment (mihailik) [inline tests/baselines/reference/literals.errors.txt]: `0o3` is not the same as `-003`.

Comment (arusakov) [inline tests/baselines/reference/literals.errors.txt]: @mihailik 
Thank you for review. I can improve reporting asap.

**Pull request merged**